src/main/java/com/clouway/push/server/PushServiceImpl.java

```diff
@@ -36,16 +35,13 @@ public void pushEvent(PushEvent event) {
   @Override
   public void pushEvent(PushEvent event, String correlationId) {
 
-    // transforming the eventType
-    if (!Strings.isNullOrEmpty(correlationId)) {
-      event.getAssociatedType().setCorrelationId(correlationId);
-    }
+    String key = event.getKey() + correlationId;
 
     String message = eventSerializer.serialize(event);
```

Comment:
Hm, something simpler would be to have the correlationId  passed as a constructor parameter directly in the PushEvent, and it's key will contain the correlationId. Example:
```java
public abstract class PushEvent  implements Serializable {

  private final String key;

  protected PushEvent(String key) {
    this.key = key;
  }

  protected PushEvent(String key, String correlationId) {
    this.key = key + correlationId;
  }

  public String getKey() {
    return key;
  }
}

// The extending class:
public class MyPushEvent extends PushEvent {

  private DataDto eventData;

  public MyPushEvent(String correlationId, DataDto dataDto) {
    super("MyPushEvent", correlationId);

    this.eventData = dataDto;
  }
}

// And the usage would be:
...
pushService.pushEvent(new MyPushEvent(correlationId, eventDataDto))
```